libpod/pod.go

```diff
@@ -666,11 +666,13 @@ func (p *Pod) AllContainersByID() ([]string, error) {
 func (p *Pod) AllContainers() ([]*Container, error) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
+	return p.allContainers()
+}
 
+func (p *Pod) allContainers() ([]*Container, error) {
 	if !p.valid {
```

Comment:
Move the valid check into AllContainers, we assume that whoever takes the lock first does the check

Reply:
done